polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -115,6 +121,13 @@ const TICK_TOO_FAR_IN_FUTURE: Tick = 20; // 10 seconds.
 const APPROVAL_DELAY: Tick = 2;
 pub(crate) const LOG_TARGET: &str = "parachain::approval-voting";
 
+// The max number of ticks we delay sending the approval after we are ready to issue the approval
+const MAX_APPROVAL_COALESCE_WAIT_TICKS: Tick = 12;
+
+// The maximum approval params we cache locally in the subsytem, so that we don't have
+// to do the back and forth to the runtime subsystem api.
+const APPROVAL_PARAMS_CACHE_SIZE: u32 = 128;
```

Comment:
We still have to pass subsystem boundaries to retrieve this information every candidate, so it might worth the optimisation, do you know if that is negligible, if it is I will remove this cache.